server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRoutes.java

```diff
@@ -182,6 +188,40 @@ private TaskIdDto userActions(Request request, Response response) throws JsonExt
         return TaskIdDto.respond(response, taskId);
     }
 
+    @POST
```

Comment:
is this really appropriated, as you are not deleting everything, but only purging > retention time messages?

Reply:
IMO `DELETE` returns 201

Here we create a related REST resource (a task) so a POST seems appropriate. The fact that this tasks actually DELETE the original endoint resources is somehow a side effect that might (hopefully) arise in the future resulting from this task execution.

IMO let's look at some work of people better than us at REST https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html : DeleteByQuery is a POST in ElasticSearch.

Agree with @rouazana on this: **it should be a POST**

Reply:
Purge & Delete API do returning tasks, and the deletion is hide by the tasks. I'm okay with POST, because, it means, a resource has been created, and that resource is a task.

Reply:
Final decision after discussion : 
* `DELETE /messages?scope=expired` for the purge operation
* delete one message doesn't change